fscrypt: also include pam module in package

ericonr opened the pull request:
The previous build style used for fscrypt, go, meant that the fscrypt PAM module, pam_fscrypt.so, wasn't included in the package. This makes PAM integration impossible if the user doesn't compile and install the module themselves.

The /etc/pam.d/fscrypt config file, which is supposed to be created by packagers to allow fscrypt to use the login password, wasn't included in the package either.

The INSTALL.msg file was added to let users know they'd still need to configure some things if they wanted a fully integrated setup. This configuration information is currently held in the Arch Wiki, where it's formatted a bit better than it is in the projects's github page. I felt it disnecessary to repeat the steps in the message, but I can copy them if you feel it's necessary, or add them to the Void Documentation.

I feel this build script has a few hacks, so I wanted to post it here to see if you have any ideas about what can be done to fix it (or if it's ok as it is).

* I added `nostrip` because it didn't build with `nopie`, and then failed during one of the lints.
* `make_install_args`: I couldn't understand what was going on with the original Makefile, but it didn't work unless I had it exactly like this. Anything I tried with setting `DESTDIR` to something made the whole thing go up in flames.
* `make_install_target`: installing only the library avoids having to remove the binary from DESTDIR, because otherwise it appears in the wrong place. The binary is then copied manually in `post_install`.
 
I don't know what Void's instace on this is, but this isn't a reproducible build. I had noticed it when traipsing through the Makefile, where the buildtime is passed as a "define" to the toolchain. The PKGBUILD used by Arch also adds a patch to make the build reproducible.

Duncaen commented:
Don't add `INSTALL.msg` this is reserved for important things.

Duncaen commented on srcpkgs/fscrypt/template:
```diff
@@ -1,14 +1,27 @@
-makedepends="pam-devel"
+revision=2
+build_style=gnu-makefile
+makedepends="go m4 pam-devel"
```
only `pam-devel` is `makedepends`, `m4` and `go` belong to `hostmakedepends`.

ericonr replied:
Just took a better look at `Manual.md`, and fixed it.

Duncaen commented on srcpkgs/fscrypt/template:
```diff
@@ -1,14 +1,27 @@
+
+make_install_args='PREFIX=${DESTDIR}/usr'
+make_install_target="install-pam"
+nostrip=yes
```
This is wrong, the pam module should still be stripped. Use `nostrip_files` to list the files that shouldn't be stripped.

ericonr replied:
Passing only the binary to `nostrip_files` fixed the issue, thanks!

Duncaen commented:
> I added nostrip because it didn't build with nopie, and then failed during one of the lints.

Then the build is wrong and doesn't respect the distributions `CFLAGS` and `LDFLAGS` if its not just the golang binaries.

ericonr commented:
I've fixed the issues you pointed out, and I will see if travis manages to build these now. Thanks for the help!

ericonr commented:
So... yeah, this seems to have quite a few issues going on ith the build. Both with 32-bit systems and arch-specific compiler flags. What do you think would be the best way of resolving this kind of stuff? Making a few arch-dependent switches? I'm going to try and cross-compile the package locally to see what I can figure out.

ericonr commented:
Well, according to voidlinux.org/packages, this package was only succesfully built for aarch64, both glibc and musl, and x86_64, only glibc. So it doesn't have the best history in cross-compilation. Right now both x86_64 targets are working (which is better than before), but aarch64 isn't. That wouldn't make for a great update, I think.

What do you think about creating a separate package for the pam module and rule? Something like pam_fscrypt? This way the packages might get too fragmented, though.

Overall I feel like adding the pam module in some way is pretty important, because it enables what I feel is the most useful part of fscrypt.

ericonr commented:
@Duncaen I have reverted back to a go build style, because the aarch64 build was failing due to Go not knowing what kind of arch it was targetting. I just defined the do_* functions inside the template file itself, to use the Makefile provided by the project.

I also had to unset LDFLAGS when building, because otherwise it errored out with an `undefined reference to main`, which I suppose came from some weird interaction with LDFLAGS that led it to try building an executable instead of a shared library. I copied this fix from the go template file itself.

Right now it's building for all 64-bit targets on CI, which is actually better than what we previously had! I hope this makes it adequate for merging :)

ericonr commented:
@Duncaen squashed and cleaned up commits, implemented your suggestions and the package is now building cleanly everywhere it can build cleanly. Any further changes?

xtraeme commented:
Andrea could you please coordinate what ypu do with the others? Thanks
